troposphere/static/js/components/common/tags/ViewTags.react.js

```diff
@@ -18,7 +18,7 @@ export default React.createClass({
 
     renderTag: function(tag) {
         return (
-        <Tag key={tag.id || tag.cid} tag={tag} renderLinks={this.props.renderLinks} />
+            <Tag key={tag.id || tag.cid} tag={tag} renderLinks={this.props.renderLinks} />
```

Comment:
That is fair. You have it reversed. cid is always, id means a response from the server